Create a docs directory under JSC and add offlineasm references to it

ast-hugger opened the pull request:
#### 6eb2566679bdbd8f10d9cb708ed9e6450ae7d173
<pre>
Create a docs directory under JSC and add offlineasm references to it
<a href="https://bugs.webkit.org/show_bug.cgi?id=301197">https://bugs.webkit.org/show_bug.cgi?id=301197</a>
<a href="https://rdar.apple.com/163127070">rdar://163127070</a>

Reviewed by Yusuke Suzuki.

This patch creates a docs/ directory under JSC root with offlineasm
instruction and register references.

No logic changes - no tests needed.

* Source/JavaScriptCore/docs/offlineasm-instruction-reference.md: Added.
* Source/JavaScriptCore/docs/offlineasm-register-reference.md: Added.

Canonical link: <a href="https://commits.webkit.org/301899@main">https://commits.webkit.org/301899@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1d56f70d0fb826353c8c025fdab975975be914d0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127392 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47040 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38177 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134456 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78947 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c4695823-ee38-47b8-a51c-bf8116c81080) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47656 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55563 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96954 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6c2fd204-5e18-40f0-a6a0-02562e9067cc) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130340 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38137 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114100 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77447 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/05d8dc3f-2cf3-4ae5-b324-daaf5b9ab4e1) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/36979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/32213 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77838 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/119429 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107996 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32621 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136939 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/125855 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/54051 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41652 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105474 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54562 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110449 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105150 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26808 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50669 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/29127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51630 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53988 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/60075 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/158893 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53222 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39727 "Passed tests") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/56679 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54981 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/JavaScriptCore/docs/offlineasm-register-reference.md:
```diff
@@ -0,0 +1,231 @@
+# JavaScriptCore Offlineasm Register Reference
+
+**NOTE: This document was largely AI-generated based on offlineasm/\*.rb sources and llint/\*.asm files. Exercise healthy skepticism.**
```
Can we also add that lots of code changes may not update this file so don't expect it to necessarily be up to date?

dhecht replied:
Could say, `Generated from 123456@main` or something like that.

kmiller68 commented on Source/JavaScriptCore/docs/offlineasm-instruction-reference.md:
```diff
@@ -0,0 +1,3950 @@
+# Offlineasm Instruction Reference
+
+> **NOTE: This document was largely AI-generated based on offlineasm/\*.rb sources and offlineasm code in llint/\*.asm files. Exercise healthy skepticism.**
```
Ditto.

ast-hugger committed:
Create a docs directory under JSC and add offlineasm references to it
https://bugs.webkit.org/show_bug.cgi?id=301197
rdar://163127070

Reviewed by Yusuke Suzuki.

This patch creates a docs/ directory under JSC root with offlineasm
instruction and register references.

No logic changes - no tests needed.

* Source/JavaScriptCore/docs/offlineasm-instruction-reference.md: Added.
* Source/JavaScriptCore/docs/offlineasm-register-reference.md: Added.

Canonical link: https://commits.webkit.org/301899@main
